tsl/src/continuous_aggs/options.c

```diff
@@ -62,12 +67,170 @@ update_materialized_only(ContinuousAgg *agg, bool materialized_only)
+{
+	List *retlist = NIL;
+	ListCell *lc;
+	Oid cagg_view_oid =
+		get_relname_relid(NameStr(agg->data.user_view_name),
+						  get_namespace_oid(NameStr(agg->data.user_view_schema), false));
+	Relation cagg_view_rel = table_open(cagg_view_oid, AccessShareLock);
+	RuleLock *cagg_view_rules = cagg_view_rel->rd_rules;
+	Assert(cagg_view_rules && cagg_view_rules->numLocks == 1);
+	RewriteRule *rule = cagg_view_rules->rules[0];
+	if (rule->event != CMD_SELECT)
+		ereport(ERROR, (errcode(ERRCODE_TS_UNEXPECTED), errmsg("unexpected rule event for view")));
+
+	Query *cagg_view_query = copyObject(linitial(rule->actions));
+	table_close(cagg_view_rel, NoLock); // lock with be released at end of txn
+	Oid mat_relid = mat_ht->main_table_relid;
+	Query *finalize_query;
+	/* the view rule has dummy old and new range table entries as the 1st and 2nd entries
+	 */
+	Assert(list_length(cagg_view_query->rtable) >= 2);
+	if (cagg_view_query->setOperations)
+	{
+		/* This corresponds to the union view.
+		 * the 3rd RTE entry has the SELECT 1 query from the union view. */
+		RangeTblEntry *finalize_query_rte = lthird(cagg_view_query->rtable);
+		if (finalize_query_rte->rtekind != RTE_SUBQUERY)
+			ereport(ERROR,
+					(errcode(ERRCODE_TS_UNEXPECTED),
+					 errmsg("unexpected rte type for view %d", finalize_query_rte->rtekind)));
+
+		finalize_query = finalize_query_rte->subquery;
+	}
+	else
+	{
+		finalize_query = cagg_view_query;
+	}
+	foreach (lc, finalize_query->groupClause)
+	{
+		SortGroupClause *cagg_gc = (SortGroupClause *) lfirst(lc);
+		TargetEntry *cagg_tle = get_sortgroupclause_tle(cagg_gc, finalize_query->targetList);
+		/* groupby clauses are columns from the mat hypertable */
+		Assert(IsA(cagg_tle->expr, Var));
+		Var *mat_var = castNode(Var, cagg_tle->expr);
+		char *mat_colname = get_attname(mat_relid, mat_var->varattno, false);
+		retlist = lappend(retlist, mat_colname);
+	}
+	return retlist;
+}
+
+static List *
+cagg_get_compression_params(ContinuousAgg *agg, Hypertable *mat_ht)
+{
+	List *defelems = NIL;
+	const Dimension *mat_ht_dim = hyperspace_get_open_dimension(mat_ht->space, 0);
+	const char *mat_ht_timecolname = NameStr(mat_ht_dim->fd.column_name);
+	DefElem *ordby = makeDefElemExtended("timescaledb",
+										 "compress_orderby",
+										 (Node *) makeString((char *) mat_ht_timecolname),
+										 DEFELEM_UNSPEC,
+										 -1);
+	defelems = lappend(defelems, ordby);
+	List *grp_colnames = cagg_find_groupingcols(agg, mat_ht);
+	if (grp_colnames)
+	{
+		ListCell *lc;
+		/* we have column names. they are guaranteed to be at most
+		 * NAMEDATALEN
+		 */
+		int seglen = ((NAMEDATALEN + 1) * list_length(grp_colnames)) + 1;
+		char *segmentby = (char *) palloc(seglen);
+		int segidx = 0;
+		foreach (lc, grp_colnames)
+		{
+			char *grpcol = (char *) lfirst(lc);
+			/* skip time dimension col if it appears in group-by list */
+			if (namestrcmp((Name) & (mat_ht_dim->fd.column_name), grpcol) == 0)
+				continue;
+			int collen = 1;
```

Comment:
This variable doesn't need an initialization anymore. Maybe move the definition to line 164?